generator/schemas/gen.go

```diff
@@ -238,16 +241,41 @@ This is not the case in the "%s' API group:
 				return
 			})
 
-			// Add the additionalProperies required to reflect the expected behavior from the K8S API,
+			// Add the additionalProperties required to reflect the expected behavior from the K8S API,
 			// (preserve-unknown-fields false by default unless on the Devfile metadata)
 			genutils.EditJSONSchema(&currentJSONSchema, func(schema *apiext.JSONSchemaProps) (newVisitor genutils.Visitor, stop bool) {
 				if schema == nil ||
-					schema.Type != "object" ||
-					schema.AdditionalProperties != nil {
+					schema.Type != "object" {
```

Comment:
Not sure I understand the logic and requirement behind this change. Could you please provide some more context (I assume I'm missing some of the history here...)

Reply:
It's described in https://github.com/devfile/api/pull/524#discussion_r664532026
Duplicating

> So, AFAIU controller-gen changed their default so now they provide empty schema with XPreserveUnknownFields, which David (I don't know reason well, probably JSON schema does not have such field) ignores during generating JSON Schema.
but schema takes over allow in JSONSchemaPropsOrBool, so it led to empty object in additional properties.
> Now, I ignore schema in additional properties if it does not define type, which is basic field.

So, some debugging.
Before was: schema is empty:
![Screenshot_20210708_123005](https://user-images.githubusercontent.com/5887312/124898891-45b79d00-dfe8-11eb-8d31-04f34503b4dc.png)
Now is, schema is set with just XPreserveUnknownFields:
![Screenshot_20210708_122728](https://user-images.githubusercontent.com/5887312/124898912-49e3ba80-dfe8-11eb-8d37-6a5241923f43.png)

Reply:
Is this change still required now that the old generation behavior has been restored by PR https://github.com/devfile/api/pull/535 ?

Reply:
Thanks. After your PR it can be reverted.